axelor-production/src/main/java/com/axelor/apps/production/service/MrpLineServiceProductionImpl.java

```diff
@@ -148,10 +149,15 @@ protected void generateManufacturingProposal(MrpLine mrpLine) throws AxelorExcep
     } else {
       BillOfMaterial defaultBillOfMaterial = billOfMaterialService.getDefaultBOM(product, company);
 
+      // The +2 adds 2 minutes to the plannedEndDateT to avoid the overflowing of the calculated
+      // plannedStartDateT on the current date time.
+      LocalDateTime maturityDateTime =
+          maturityDate.isEqual(LocalDate.now())
+              ? maturityDate.atTime(LocalTime.now().plusMinutes(2))
+              : maturityDate.atStartOfDay();
```

Comment:
for testing purpose, please use appBaseService.getTodayDateTime(company) to get the current date and time.